mysql/util.go

```diff
@@ -452,21 +453,43 @@ func ErrorEqual(err1, err2 error) bool {
 	return e1.Error() == e2.Error()
 }
 
+func compareSubVersion(typ, a, b, aFull, bFull string) (int, error) {
+	var aNum, bNum int
+	var err error
+
+	if aNum, err = strconv.Atoi(a); err != nil {
+		return 0, fmt.Errorf("cannot parse %s version %s of %s", typ, a, aFull)
+	}
+	if bNum, err = strconv.Atoi(b); err != nil {
+		return 0, fmt.Errorf("cannot parse %s version %s of %s", typ, b, bFull)
+	}
+
+	return cmp.Compare(aNum, bNum), nil
+}
+
 func CompareServerVersions(a, b string) (int, error) {
```

Comment:
Please add some comments to explain the behaviour of this function. Like it will ignore bytes after `-`, it assumes input string must in format `{major}.{minor}.{patch}`

Reply:
Commented (with description of extended wildcard behavior) & added more tests